algorithms/src/snark/marlin/ahp/indexer/circuit.rs

```diff
@@ -53,11 +76,44 @@ pub struct Circuit<F: PrimeField, MM: MarlinMode> {
 
     pub fft_precomputation: FFTPrecomputation<F>,
     pub ifft_precomputation: IFFTPrecomputation<F>,
+    pub(crate) _mode: PhantomData<MM>,
+    pub(crate) id: CircuitId,
+}
 
-    pub(crate) mode: PhantomData<MM>,
+impl<F: PrimeField, MM: MarlinMode> Eq for Circuit<F, MM> {}
+impl<F: PrimeField, MM: MarlinMode> PartialEq for Circuit<F, MM> {
```

Comment:
do we ever plan to have the `Circuit` as a key in some map? if it's a possibility, we should add a [matching `std::hash::Hash` impl](https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq)

Reply:
If it's not needed now, I would avoid adding dead code right?

Reply:
as long as there is no plan to use it as such, indeed :+1: